### PR TITLE
Homework stack done

### PR DESCRIPTION
**Pull request by goharminasyan**

### CODE REVIEW

File: src/stack/Stack.java [hunk 1]
```diff
@@ -1,56 +1,92 @@
...
+        data = new int[size];
+        this.tos = -1;
+    }
 
-	}
+    /**
+     * -* Add element into stack
+     */
+    public void push(int value) {
+        if (tos == 15) {
```
Comment (Karlen999) [inline src/stack/Stack.java:39]: You can write **tos == data.length - 1**

File: src/stack/Stack.java [hunk 2]
```diff
@@ -1,56 +1,92 @@
...
-	 * @return
-	 */
-	public int pop() {
+    /**
+     * Get element from the stack
+     *
+     * @return
+     */
+    public int pop() {
+        if (tos < 0) {
```
Comment (Karlen999) [inline src/stack/Stack.java:51]: You can use **isEmpty()** method

File: src/stack/Stack.java [hunk 3]
```diff
@@ -1,56 +1,92 @@
...
-	 * Increase the stack capacity if there is not enough space to add additional items
-	 */
-	private void ensureCapacity() {
-
-	}
+        if (tos >= DEFAULT_SIZE) {
+            int newCapacity = (DEFAULT_SIZE * 2) + 1;
+            data[DEFAULT_SIZE] = data[newCapacity];
+        }
+    }
```
Comment (Karlen999) [inline src/stack/Stack.java:91]: done

Comment (goharminasyan) [reply src/stack/Stack.java:91]: In this method I have such a problem․ How can I correct?
Non-static method 'ensureCapacity()' cannot be referenced from a static context։18

Comment (Karlen999) [reply src/stack/Stack.java:91]: For example like this

private void ensureCapacity() {
		int[] temp = new int[data.length + DEFAULT_SIZE];
		System.arraycopy(data, 0, temp, 0, data.length);
		data = temp;
	}

**Comment by Karlen999** [comment]
reviewed

**Pull request closed without merging**